### PR TITLE
feat: add send to flomo extension

### PR DESCRIPTION
**Pull request by ahonn**
## Description
Add the send-to-flomo plugin to quickly record thoughts to [flomo](flomoapp.com) in raycast

## Type of change

- New extension

## Screencast

3s6Qrg

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I checked that extension's icon has size of `512x512`

### CODE REVIEW

**Comment by designedbyclu** [comment]
I was planing to do this send to flomo so. Thanks for doing it first.

File: extensions/send-to-flomo/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Send to flomo
+
+Raycast extension for send MEMO to flomo
```
Comment (pernielsentikaer) [inline extensions/send-to-flomo/README.md:3]:
```suggestion
Raycast extension for sending MEMOs to flomo
```

File: extensions/send-to-flomo/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+## Setup Extension
+1. Copy and set your MEMO API from [flomo](https://flomoapp.com/mine?source=incoming_webhook)
```
Comment (pernielsentikaer) [inline extensions/send-to-flomo/README.md:6]:
```suggestion
1. Copy and set your MEMO API URL from [flomo](https://flomoapp.com/mine?source=incoming_webhook)
```

File: extensions/send-to-flomo/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+![set your MEMO API](https://ahonn-me.oss-cn-beijing.aliyuncs.com/images/zwDSmI.png)
```
Comment (pernielsentikaer) [inline extensions/send-to-flomo/README.md:8]:
```suggestion
![Set your MEMO API URL](https://ahonn-me.oss-cn-beijing.aliyuncs.com/images/zwDSmI.png)
```

File: extensions/send-to-flomo/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+2. Write your thoughts and send to flomo (cmd + enter)
```
Comment (pernielsentikaer) [inline extensions/send-to-flomo/README.md:10]:
```suggestion
2. Write your thoughts and send the MEMO to flomo (cmd + enter)
```

File: extensions/send-to-flomo/README.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+![send MEMO](https://ahonn-me.oss-cn-beijing.aliyuncs.com/images/JVmqBL.png)
```
Comment (pernielsentikaer) [inline extensions/send-to-flomo/README.md:12]:
```suggestion
![Send MEMO](https://ahonn-me.oss-cn-beijing.aliyuncs.com/images/JVmqBL.png)
```

File: extensions/send-to-flomo/package.json [hunk 6]
```diff
@@ -0,0 +1,53 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "send-to-flomo",
+  "title": "Send to flomo",
+  "description": "Raycast extension for send MEMO to flomo",
```
Comment (pernielsentikaer) [inline extensions/send-to-flomo/package.json:5]:
```suggestion
  "description": "Raycast extension for sending MEMOs to flomo",
```

File: extensions/send-to-flomo/package.json [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  "icon": "command-icon.png",
+  "author": "ahonn",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "send-to-flomo",
+      "title": "Send to flomo",
```
Comment (pernielsentikaer) [inline extensions/send-to-flomo/package.json:12]:
```suggestion
      "title": "Send MEMO",
```

File: extensions/send-to-flomo/package.json [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+      "subtitle": "What's in your mind",
```
Comment (pernielsentikaer) [inline extensions/send-to-flomo/package.json:13]:
```suggestion
      "subtitle": "Flomo",
```

File: extensions/send-to-flomo/package.json [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+      "description": "Send your thoughts to flomo",
+      "mode": "view"
+    }
+  ],
+  "preferences": [
+    {
+      "name": "api",
+      "title": "MEMO API",
```
Comment (pernielsentikaer) [inline extensions/send-to-flomo/package.json:21]:
```suggestion
      "title": "MEMO API URL",
```

File: extensions/send-to-flomo/package.json [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+      "description": "Your MEMO API",
```
Comment (pernielsentikaer) [inline extensions/send-to-flomo/package.json:22]:
```suggestion
      "description": "Your MEMO API URL",
```

File: extensions/send-to-flomo/package.json [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+      "type": "textfield",
+      "required": true
+    },
+    {
+      "name": "defaultTag",
+      "title": "Default tag",
```
Comment (pernielsentikaer) [inline extensions/send-to-flomo/package.json:28]:
```suggestion
      "title": "Default tag(s)",
```

File: extensions/send-to-flomo/package.json [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+      "description": "The default tag that comes with each send",
```
Comment (pernielsentikaer) [inline extensions/send-to-flomo/package.json:29]:
```suggestion
      "description": "The default tag(s) that comes with each MEMO — separated by space",
```

File: extensions/send-to-flomo/src/send-to-flomo.tsx [hunk 13]
```diff
@@ -0,0 +1,49 @@
+import axios from "axios";
+import { useState } from "react";
+import {
+  getPreferenceValues,
+  ActionPanel,
+  Form,
+  FormValue,
+  SubmitFormAction,
+  showToast,
+  ToastStyle,
```
Comment (pernielsentikaer) [inline extensions/send-to-flomo/src/send-to-flomo.tsx:10]:
```suggestion
  ToastStyle
```

File: extensions/send-to-flomo/src/send-to-flomo.tsx [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+      content += ` ${preferences.defaultTag}`;
+    }
+
+    const response = await axios.post(preferences.api, { content });
+    if (response.data?.code === 0) {
+      showToast(ToastStyle.Success, "Success");
+      setText('');
+    } else {
+      showToast(ToastStyle.Failure, "Failed");
+    }
```
Comment (pernielsentikaer) [inline extensions/send-to-flomo/src/send-to-flomo.tsx:35]:
```suggestion
if(content !== "") {

      if (preferences.defaultTag) {
        content += ` ${preferences.defaultTag}`;
      }

      showToast(ToastStyle.Animated, "Sending")

      const response = await axios.post(preferences.api, { content });

      if (response.data?.code === 0) {
        showToast(ToastStyle.Success, "Success", "Successfully send MEMO");
        setText('');
      } else {
        showToast(ToastStyle.Failure, "Failed", "Check that API MEMO URL is correct");
      }
    }
    else {
      showToast(ToastStyle.Failure, "Failed", "MEMO can't be left empty");
    }
```

Comment (pernielsentikaer) [reply extensions/send-to-flomo/src/send-to-flomo.tsx:35]: Added a toast spinner to indicate that it's being processed, it takes up to a few seconds to send, and this way we improve the UX. Also added an error if nothing is written.

File: extensions/send-to-flomo/src/send-to-flomo.tsx [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <SubmitFormAction title="Send MEMO" onSubmit={handleSubmit} />
+        </ActionPanel>
+      }
+    >
+      <Form.TextArea id="memo" title="MEMO" placeholder="What's in your mind?" value={text} onChange={setText} />
```
Comment (pernielsentikaer) [inline extensions/send-to-flomo/src/send-to-flomo.tsx:46]:
```suggestion
      <Form.TextArea id="memo" title="MEMO" placeholder="What's on your mind?" value={text} onChange={setText} />
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @ahonn 👋 

What an awesome extension, very nice 🚀

Left a few suggestions that we need to address before It's ready to get merged.

The provided logo in the extension is not provided in 512x512 pixels. Please modify this according to [Extension Icon guideline](https://developers.raycast.com/basics/prepare-an-extension-for-store#extension-icon)

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by ahonn** [comment]
@pernielsentikaer Thanks for the suggestion, I've fixed it. Then I also added a Form.TextField to edit/add labels, please review it again.

File: extensions/send-to-flomo/package.json [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+      "name": "api",
+      "title": "MEMO API URL",
+      "description": "Your MEMO API URL",
+      "type": "textfield",
+      "required": true
+    },
+    {
+      "name": "defaultTags",
+      "title": "Default tag(s)",
+      "description": "The default tag(s) that comes with each MEMO — separated by space",
```
Comment (pernielsentikaer) [inline extensions/send-to-flomo/package.json:29]:
```suggestion
      "title": "Default tags",
      "description": "The default tags that comes with each MEMO — separated by space",
```

My bad, this sounds better 😄

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @ahonn 👋 

Thanks for the quick update 💯 

I can see that our linter is not happy about the image, I'll check and get back to you asap since it looks correct to me.

Left a single suggestion, my bad, can we change this meanwhile?

**Comment by ahonn** [comment]
@designedbyclu You may want to try the [Send to flomo](https://www.icloud.com/shortcuts/db96b580335d4c23a5ba6416d98cb8d7) shortcut on Mac (macOS Monterey)

File: extensions/send-to-flomo/package.json (hunk 16 above)
Comment (ahonn) [reply extensions/send-to-flomo/package.json]: Agree, I updated this command description

**Review by pernielsentikaer — APPROVED** [review]
@ahonn 👋 Another quick update 💣  💯 

I can't find anything else I want to have changed, I'm going to accept it, and my colleague will check one last time before merging it 💪 

Thanks again for the contribution 😄

File: extensions/send-to-flomo/package.json [hunk 17]
```diff
@@ -0,0 +1,53 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "send-to-flomo",
+  "title": "Send to flomo",
+  "description": "Raycast extension for sending MEMOs to flomo",
```
Comment (fe9lix) [inline extensions/send-to-flomo/package.json:5]:
```suggestion
  "description": "Send your MEMOs to flomo",
```

File: extensions/send-to-flomo/src/send-to-flomo.tsx [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+
+    if (content !== '') {
+      if (tags) {
+        content += ` ${values.tags}`;
+      }
+      showToast(ToastStyle.Animated, 'Sending');
+      const response = await axios.post(preferences.api, { content });
+
+      if (response.data?.code === 0) {
+        showToast(ToastStyle.Success, 'Success', 'Successfully send MEMO');
```
Comment (fe9lix) [inline extensions/send-to-flomo/src/send-to-flomo.tsx:34]:
```suggestion
        showToast(ToastStyle.Success, 'Success', 'Successfully sent MEMO');
```

**Review by fe9lix — CHANGES_REQUESTED** [review]
Just left some more minor typos to be addressed.

**Comment by ahonn** [comment]
@fe9lix  The typos mentioned have been updated

**Review by fe9lix — APPROVED** [review]

**Pull request merged**